src/libraries/Microsoft.Extensions.Hosting/src/Internal/Host.cs

```diff
@@ -52,6 +52,14 @@ public async Task StartAsync(CancellationToken cancellationToken = default)
             {
                 // Fire IHostedService.Start
                 await hostedService.StartAsync(combinedCancellationToken).ConfigureAwait(false);
+
+                if (hostedService is BackgroundService backgroundService)
+                {
+                    _ = backgroundService.ExecuteTask.ContinueWith(t =>
```

Comment:
@stephentoub might disagree with me but I'd prefer a local async function than continue with:

```C#
async Task HandleException()
{
    try
    {
        await backgroundService.ExecuteTask;
    }
    catch (Exception ex)
    {
        _logger.BackgroundServiceFaulted(t.Exception);
    }
}
```

Reply:
No disagreement.  In general I'd suggest using async as well unless profiling demonstrates there's an advantage to ContinueWith (there can be but the situations are pretty limited).  Especially if this isn't a very frequent operation, async/await all the way.

Reply:
Fixed.